File: partition.py
# TC: O(NlogN)
# SC: O(1)

class Solution:
    def arrayPairSum(self, nums: List[int]) -> int:
        
        nums = sorted(nums)
        s = 0
        
        i = 0
        while i < len(nums)-1:
            s += nums[i]
            i += 2
            
        return s

File: partition.java
class Solution {
    public int arrayPairSum(int[] nums) {
        
        Arrays.sort(nums);
        int sum = 0;
        for(int i=0;i<nums.length;i+=2){
            sum+= nums[i];
        }
        return sum;
    }
}
// Time Complexity: O(nlogn)
// Space complexity: O(logn)
